### ISSUE TITLE
redis stops installation of mastodon

### ISSUE THREAD

**Issue by halogeno** [mastodon/mastodon#14475]
On a fresh ubuntu 18.04 LTS server I go through all steps until:

...
Are you using Docker to run Mastodon? no

PostgreSQL host: /var/run/postgresql
PostgreSQL port: 5432
Name of PostgreSQL database: mastodon_production
Name of PostgreSQL user: mastodon
Password of PostgreSQL user: 
Database configuration works! 🎆

Redis host: localhost
Redis port: 6379
Redis password: 
Redis connection could not be established with this configuration, try again.
Error connecting to Redis on localhost:6379 (Errno::ECONNREFUSED)
Try again?
...

I've tried it several times on a fresh updated 18.04 server and still the same problem.

**Comment by Gargron** [comment]
Is redis running?

**Comment by halogeno** [comment]
No.

alpha555:~# systemctl status redis-server.service
● redis-server.service - Advanced key-value store
   Loaded: loaded (/lib/systemd/system/redis-server.service; disabled; vendor preset: enabled)
   Active: activating (start) since Sat 2020-08-01 07:48:16 CEST; 1min 16s ago
     Docs: http://redis.io/documentation,
           man:redis-server(1)
  Process: 16782 ExecStart=/usr/bin/redis-server /etc/redis/redis.conf (code=exited, status=0/SUCCESS)
    Tasks: 0 (limit: 4915)
   CGroup: /system.slice/redis-server.service

Aug 01 07:48:16 alpha555 systemd[1]: Starting Advanced key-value store...
Aug 01 07:48:17 alpha555 systemd[1]: redis-server.service: Can't open PID file /var/run/redis/redis-server.pid (yet?) after start: No such file or directory
alpha555:~# ps ax | grep redis
16829 pts/0    S+     0:00 grep redis
alpha555:~#

**Comment by halogeno** [comment]
`alpha555:~# systemctl status redis-server.service
● redis-server.service - Advanced key-value store
   Loaded: loaded (/lib/systemd/system/redis-server.service; disabled; vendor preset: enabled)
   Active: activating (start) since Sat 2020-08-01 07:48:16 CEST; 1min 16s ago
     Docs: http://redis.io/documentation,
           man:redis-server(1)
  Process: 16782 ExecStart=/usr/bin/redis-server /etc/redis/redis.conf (code=exited, status=0/SUCCESS)
    Tasks: 0 (limit: 4915)
   CGroup: /system.slice/redis-server.service

Aug 01 07:48:16 alpha555 systemd[1]: Starting Advanced key-value store...
Aug 01 07:48:17 alpha555 systemd[1]: redis-server.service: Can't open PID file /var/run/redis/redis-server.pid (yet?) after start: No such file or directory
alpha555:~# ps ax | grep redis
16829 pts/0    S+     0:00 grep redis
alpha555:~#`

**Comment by halogeno** [comment]
No.

alpha555:~# systemctl status redis-server.service
● redis-server.service - Advanced key-value store
   Loaded: loaded (/lib/systemd/system/redis-server.service; disabled; vendor preset: enabled)
   Active: activating (start) since Sat 2020-08-01 07:48:16 CEST; 1min 16s ago
     Docs: http://redis.io/documentation,
           man:redis-server(1)
  Process: 16782 ExecStart=/usr/bin/redis-server /etc/redis/redis.conf (code=exited, status=0/SUCCESS)
    Tasks: 0 (limit: 4915)
   CGroup: /system.slice/redis-server.service

Aug 01 07:48:16 alpha555 systemd[1]: Starting Advanced key-value store...
Aug 01 07:48:17 alpha555 systemd[1]: redis-server.service: Can't open PID file /var/run/redis/redis-server.pid (yet?) after start: No such file or directory
alpha555:~# ps ax | grep redis
16829 pts/0    S+     0:00 grep redis
alpha555:~#

**Comment by Gargron** [comment]
> Can't open PID file /var/run/redis/redis-server.pid (yet?) after start: No such file or directory

I don't know what's going on there, never encountered this on Ubuntu 18.04, but essentially you need to ensure that Redis works before you can attempt to do anything with Mastodon.

Maybe something to do with IPv6 like in this issue? https://stackoverflow.com/questions/50668845/cannot-install-redis-server-on-ubuntu-18-04 Redis only needs to listen on IPv4 127.0.0.1.

**Comment by halogeno** [comment]
It worked! (but what a strange behavior. Maybe because my server had only a private IPv6-Adress)

after

apt install -y \
  imagemagick ffmpeg libpq-dev libxml2-dev libxslt1-dev file git-core \
  g++ libprotobuf-dev protobuf-compiler pkg-config nodejs gcc autoconf \
  bison build-essential libssl-dev libyaml-dev libreadline6-dev \
  zlib1g-dev libncurses5-dev libffi-dev libgdbm5 libgdbm-dev \
  nginx redis-server redis-tools postgresql postgresql-contrib \
  certbot python-certbot-nginx yarn libidn11-dev libicu-dev libjemalloc-dev

I have edited /etc/redis/redis.conf like your suggested stackoverflow-article from

bind 127.0.0.1 ::1

to

bind 127.0.0.1

and after a following

systemctl restart redis-server.service

I have finished the installation.

Thanks!
